src/components/TodoFilter/TodoFilter.tsx

```diff
@@ -1,30 +1,79 @@
-export const TodoFilter = () => (
-  <form className="field has-addons">
-    <p className="control">
-      <span className="select">
-        <select data-cy="statusSelect">
-          <option value="all">All</option>
-          <option value="active">Active</option>
-          <option value="completed">Completed</option>
-        </select>
-      </span>
-    </p>
-
-    <p className="control is-expanded has-icons-left has-icons-right">
-      <input
-        data-cy="searchInput"
-        type="text"
-        className="input"
-        placeholder="Search..."
-      />
-      <span className="icon is-left">
-        <i className="fas fa-magnifying-glass" />
-      </span>
-
-      <span className="icon is-right" style={{ pointerEvents: 'all' }}>
-        {/* eslint-disable-next-line jsx-a11y/control-has-associated-label */}
-        <button data-cy="clearSearchButton" type="button" className="delete" />
-      </span>
-    </p>
-  </form>
-);
+import { ChangeEvent, useCallback, useState } from 'react';
+import { Filter } from '../../enums/Filter';
+
+type Props = {
+  onFilterChange: (newFilter: Filter) => void;
+  onQueryChange: (newQuery: string) => void;
+};
+
+export const TodoFilter = ({ onFilterChange, onQueryChange }: Props) => {
+  const [query, setQuery] = useState('');
+
+  const handleFilterChange = useCallback(
+    (event: ChangeEvent<HTMLSelectElement>) => {
+      const newFilter = event.target.value as Filter;
```

Comment:
in this case you can use `event.currentTarget.value` and you don't need to use assertion type